source/OrderedTypes/SupLattice.lagda

```diff
@@ -0,0 +1,286 @@
+                             → (order-of L (join-of L U) u') holds
+join-is-least-upper-bound-of L U = pr₂ (join-is-lub-of L U)
+
+sethood-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-set ⟨ L ⟩
+sethood-of L =
+  type-with-prop-valued-refl-antisym-rel-is-set
+   (λ x → λ y → order-of L x y holds)
+   (λ x → λ y → holds-is-prop (order-of L x y))
+   (λ x → reflexivity-of L x)
+   (λ x → λ y → antisymmetry-of L)
+
+\end{code}
+
+We now define monotone endomaps on a sup-lattice. This notion is not too
+restrictive as our interest is with fixed points.
+
+\begin{code}
+
+module _ {𝓤 𝓦 𝓥 : Universe} (L : Sup-Lattice 𝓤 𝓦 𝓥) where
+
+ is-monotone : (f : ⟨ L ⟩ → ⟨ L ⟩) → 𝓤 ⊔ 𝓦  ̇
+ is-monotone f = (x y : ⟨ L ⟩) → (x ≤⟨ L ⟩ y) holds → (f x ≤⟨ L ⟩ f y) holds
+
+\end{code}
+
+We now show that when one subset contains another the join of their total
+spaces are ordered as expected. 
+
+\begin{code}
+
+module _ {𝓤 𝓦 𝓥 : Universe}
+         (L : Sup-Lattice 𝓤 𝓦 𝓥)
+         {A : 𝓥  ̇}
+         (m : A → ⟨ L ⟩)
+          where
+
+ open Joins (order-of L)
+
+ joins-preserve-containment : {P : 𝓟 {𝓥} A} {Q : 𝓟 {𝓥} A}
+                            → P ⊆ Q
+                            → ((⋁⟨ L ⟩ (𝕋 P , m ∘ 𝕋-to-carrier P))
+                             ≤⟨ L ⟩ (⋁⟨ L ⟩ (𝕋 Q , m ∘ 𝕋-to-carrier Q))) holds
+ joins-preserve-containment {P} {Q} C =
+   (join-is-least-upper-bound-of L (𝕋 P , m ∘ 𝕋-to-carrier P))
+    (⋁⟨ L ⟩ (𝕋 Q , m ∘ 𝕋-to-carrier Q) ,
+    (λ (b , b-in-P)
+      → (join-is-upper-bound-of L (𝕋 Q , m ∘ 𝕋-to-carrier Q))
+        (b , C b b-in-P)))
+
+\end{code}
+
+We now show if a type is small and has a map to the carrier then it has a join.
+
+\begin{code}
+
+module _ {𝓤 𝓦 𝓥 𝓣 : Universe}
+         (L : Sup-Lattice 𝓤 𝓦 𝓥)
+         {T : 𝓣  ̇}
+         (m : T → ⟨ L ⟩)
+         (t : T is 𝓥 small)
+          where
+ private 
+  T' : 𝓥  ̇
+  T' = (resized T) t
+
+  T'-≃-T : T' ≃ T
+  T'-≃-T = resizing-condition t
+
+  T'-to-T : T' → T
+  T'-to-T = ⌜ T'-≃-T ⌝
+ 
+  T'-to-T-is-equiv : is-equiv T'-to-T
+  T'-to-T-is-equiv = ⌜ T'-≃-T ⌝-is-equiv
+
+  T-to-T' : T → T'
+  T-to-T' =  ⌜ T'-≃-T ⌝⁻¹
+
+  T'-to-T-has-section : has-section T'-to-T
+  T'-to-T-has-section = equivs-have-sections T'-to-T T'-to-T-is-equiv
+
+  T'-to-T-is-section : is-section T'-to-T
+  T'-to-T-is-section = equivs-are-sections T'-to-T T'-to-T-is-equiv
+
+  section-T'-to-T : T'-to-T ∘ T-to-T' ∼ id
+  section-T'-to-T = section-equation T'-to-T T'-to-T-has-section
+
+  retraction-T'-to-T : T-to-T' ∘ T'-to-T ∼ id
+  retraction-T'-to-T = inverses-are-retractions' T'-≃-T
+
+  T'-inclusion : T' → ⟨ L ⟩
+  T'-inclusion = m ∘ T'-to-T
+
+  s : ⟨ L ⟩
+  s = ⋁⟨ L ⟩ (T' , T'-inclusion)
+
+ open Joins (order-of L)
+
+ sup-of-small-fam-is-lub : (s is-lub-of (T , m)) holds
+ sup-of-small-fam-is-lub = (I , III)
+  where
+   I : (s is-an-upper-bound-of (T , m)) holds
+   I t = II
+    where
+     II : (m t ≤⟨ L ⟩ s) holds 
+     II = transport (λ z → (m z ≤⟨ L ⟩ s) holds)
```

Comment:
FYI: For `transport` the convention is to always use `-` for the variable (which you denote by `z` here).